### PR TITLE
Settings interface update

### PR DESCRIPTION
**Pull request by stephywells**
This isn't ready to be merged yet. It's just here so we can start a conversation. It's labeled as "in progress" until we're ready to merge it.

@theRealRizeo This is looking so much better! Nice job on this! I made a few changes to simplify some of the css.

Now, before we can merge this, there are still a few things left to do:
- [x] Add an icon for the Miscellaneous nav. @semelyak it looks like we're missing this one.
- [x] Apply this same layout to more pages. Here's the design https://www.figma.com/file/hey3bHjXZhJpGGt277mHbg/Business_directory?node-id=807%3A34993
- [x] I'm having a really hard time finding the css I'm looking for. The organization isn't straightforward, and all the &-whatever makes it really hard to search for something. I think we can use the wpbdp prefix separate, and anything that starts with : . But other than that, this needs to be organized a bit differently to make it easy to find with a search.
- [x] Please switch all the spaces to tabs in these new css files.

### CODE REVIEW

**Comment by theRealRizeo** [comment]
Hey @stephywells thanks, this makes it easier :) Will work on the changes

**Comment by theRealRizeo** [comment]
image

Icons fix pushed. Themes add a section called "Licenses" and need to be aligned as well. Later on I think we can move this to one license area

**Comment by semelyak** [comment]
@stephywells , @theRealRizeo 
Added missed icons to Figma. 
https://www.figma.com/file/hey3bHjXZhJpGGt277mHbg/Business_directory?node-id=1158%3A54264

**Comment by theRealRizeo** [comment]
@stephywells done some changes for views, please review :) . This includes categories and tags. 
Loading dialog using `jquery-ui-dialog` as Formidable and used most of its CSS in the files from the plugin

File: assets/css/less/admin/modal.less [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.wpbdp-admin-dialog {
```
Comment (stephywells) [inline assets/css/less/admin/modal.less:1]: This modal needs another look to make it look like the design. I'm fine with not adding the second button, but the style needs to look a lot closer (lines go to the edges, spacing...)

File: includes/helpers/class-app.php [hunk 2]
```diff
@@ -313,20 +313,20 @@ public static function array_to_html_params( $atts ) {
 	/**
 	 * @since 5.9.2
 	 */
-	public static function show_logo( $size ) {
-		echo self::kses( self::svg_logo( $size ), 'all' ); // WPCS: XSS ok.
+	public static function show_logo( $size, $extra_class = false, $round = false ) {
```
Comment (stephywells) [inline includes/helpers/class-app.php:316]: It looks like it would be better to add a new function for the round version, instead of the extra complexity with all the arguments.

File: includes/admin/helpers/functions/layout.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+/**
+ * Prints out all settings sections added to a particular settings page.
+ *
+ * @link https://developer.wordpress.org/reference/functions/do_settings_sections/
+ *
+ * @param string $page
+ *
+ * @since x.x
+ */
+function wpbdp_admin_do_settings_sections( $page ) {
```
Comment (stephywells) [inline includes/admin/helpers/functions/layout.php:17]: Looks like these functions should be in a class. Maybe the admin pages class would make sense?

**Review by stephywells — COMMENTED** [review]
This is some great progress. There's a ton here, so this will take a while to go through all the big diffs. But here's a start.

It would also be great to go through and check spacing between items in the settings and compare to the design.

File: includes/admin/class-education.php [hunk 4]
```diff
@@ -99,12 +99,15 @@ private static function tips() {
 			'discounts' => array(
 				'requires' => 'discount-codes',
 				'tip'      => 'Offer discount & coupon codes to your paid listing customers.',
-				'cta'      => 'Upgrade to Pro.',
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply includes/admin/class-education.php:102]: I was using the Pro version of the plugin, but the message still says "Upgrade to Pro". Made no sense to me as the addon is visible in the Modules section and a redirect there would make more sense

File: includes/admin/helpers/functions/layout.php (hunk 3 above)
Comment (theRealRizeo) [reply includes/admin/helpers/functions/layout.php:17]: Agreed :)

File: includes/admin/upgrades/class-manual-upgrade-helper.php [hunk 5]
```diff
@@ -149,7 +149,12 @@ private function is_configured() {
     }
 
     public function upgrade_page() {
-        echo wpbdp_admin_header( __( 'Business Directory - Manual Upgrade', 'business-directory-plugin' ), 'manual-upgrade', null, false );
+		echo wpbdp_admin_header(
+			array(
+				'title' => __( 'Business Directory - Manual Upgrade', 'business-directory-plugin' ),
+				'id'    => 'manual-upgrade'
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply includes/admin/upgrades/class-manual-upgrade-helper.php]: Added :)

File: includes/utils.php [hunk 6]
```diff
@@ -530,6 +530,31 @@ function wpbdp_sanitize_value( $sanitize, &$value ) {
...
+ * This sanitizes multiple html classes that are an array of classes or a string of them separated by a delimiter.
+ *
+ * @param string|array $classes The html classes.
+ * @param string $sep The seperator of the strings. Defaults to ' '.
+ *
+ * @since x.x
+ *
+ * @return string
+ */
+function wpbdp_sanitize_html_classes( $classes, $sep = ' ' ) {
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply includes/utils.php]: Moved to App_Helper class

File: templates/admin/themes-delete-confirm.tpl.php [hunk 7]
```diff
@@ -1,4 +1,13 @@
-<?php echo wpbdp_admin_header( __( 'Delete Directory Theme', 'business-directory-plugin' ), 'theme-delete' ); ?>
+<?php
+wpbdp_admin_header(
+    array(
+        'title'   => __( 'Delete Directory Theme', 'business-directory-plugin' ),
+		'id'      => 'themes-delete',
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply templates/admin/themes-delete-confirm.tpl.php]: I'll fix in all files changed

File: assets/css/less/admin/modal.less (hunk 1 above)
Comment (theRealRizeo) [reply assets/css/less/admin/modal.less:1]: Will work on those. The modal is currently rendered using the existing forms in he category and tag pages. Will improve the styling :)

Comment (theRealRizeo) [reply assets/css/less/admin/modal.less:1]: Modal styling has been updated

File: includes/helpers/class-app.php (hunk 2 above)
Comment (theRealRizeo) [reply includes/helpers/class-app.php]: Done

**Comment by theRealRizeo** [comment]
@stephywells got some questions about some views and elements:

1. The file upload field under https://www.figma.com/file/hey3bHjXZhJpGGt277mHbg/Business_directory?node-id=320%3A7232 on the plugin is loaded via an iframe. It will require changing the element to a non-iframe and adding some custom scripts for it (I did the styling for CSV import)
2. The Form Fields https://www.figma.com/file/hey3bHjXZhJpGGt277mHbg/Business_directory?node-id=610%3A8148 styling will require changing from table to divs
3. Should I leave the input paddings as per the current changes?

**Comment by stephywells** [comment]
1. I'm not sure what file upload you're referring to. Is that the right link? Or maybe you can include a screenshot?
2. Let's hold off on that part. We can work on that in a separate branch.
3. Yeah, let's leave the input fields as they are now. The padding was causing a lot of weird issues, so I don't think it's worth changing.

Another thing I noticed... It looks like we lost error messages and success messages. We'll need to recover those and get them to show in the right spot.

**Comment by theRealRizeo** [comment]
@stephywells the file upload designs are these :
Screenshot 2022-01-07 114959

Current page loads using an iframe as:
Screenshot 2022-01-07 115134

Will check on the error and success messages

**Comment by theRealRizeo** [comment]
Hey Steph, I set the messages to show in the notification area. There is CSS that was added to hide the messages. There is no current design on how other notifications should work, as of now it just shows "Settings Saved" . 
Screenshot 2022-01-07 132227

Perhaps for small notices, a small floating notification would be good like in the Gutenberg editor, positioning can be top center of the page

Screenshot 2022-01-07 132548

File: templates/admin/form-fields.tpl.php [hunk 8]
```diff
@@ -1,39 +1,29 @@
...
-                admin_url( 'admin.php?page=wpbdp_admin_formfields&action=previewform' )
-            ),
-        ),
-        array(
-            _x( 'Manage Theme Tags', 'form-fields admin', 'business-directory-plugin' ),
-            esc_url(
-                admin_url( 'admin.php?page=wpbdp_admin_formfields&action=updatetags' )
-            ),
-        ),
+        __( 'Preview Form', 'business-directory-plugin' )      => admin_url( 'admin.php?page=wpbdp_admin_formfields&action=previewform' ),
```
Comment (stephywells) [inline templates/admin/form-fields.tpl.php:8]: Should these be in the new array format for buttons?

File: includes/utils.php [hunk 9]
```diff
@@ -11,33 +11,33 @@
...
-     * @since 5.0
-     */
-    public static function sort_by_property( &$array, $prop ) {
-        self::$property = $prop;
-
-        uasort( $array, array( 'WPBDP_Utils', 'sort_by_property_callback' ) );
-
-        self::$property = null;
-    }
+	/**
```
Comment (stephywells) [inline includes/utils.php:14]: This diff is too big. Can you please back out the phpcs for lines that aren't being changed?

File: includes/admin/settings/class-settings-bootstrap.php [hunk 10]
```diff
@@ -4,36 +4,50 @@
...
-        wpbdp_register_settings_group( 'payment', _x( 'Payment', 'settings', 'business-directory-plugin' ) );
-        wpbdp_register_settings_group( 'payment/main', _x( 'General Settings', 'settings', 'business-directory-plugin' ), 'payment' );
-
-        wpbdp_register_settings_group( 'appearance', _x( 'Appearance', 'settings', 'business-directory-plugin' ) );
-
-        wpbdp_register_settings_group( 'modules', __( 'Module Settings', 'business-directory-plugin' ) );
-
-		wpbdp_register_settings_group( 'misc', __( 'Miscellaneous', 'business-directory-plugin' ) );
-    }
+	public static function register_initial_groups() {
```
Comment (stephywells) [inline includes/admin/settings/class-settings-bootstrap.php]: The diff in this file is too big also. Can you please revert the tabs/spaces for lines that aren't getting changed?

File: includes/admin/class-admin.php [hunk 11]
```diff
@@ -22,97 +22,100 @@
 
 if ( ! class_exists( 'WPBDP_Admin' ) ) {
 
-    /**
-     * Class WPBDP_Admin
-     */
-    class WPBDP_Admin {
+	/**
```
Comment (stephywells) [inline includes/admin/class-admin.php:25]: Let's back out the extra phpcs changes here too

**Comment by stephywells** [comment]
Thanks for clearing up the upload iframe thing. Can we load the css with the file upload styling in the iframe? I'd prefer to avoid functionality changes in this PR if possible.

For the notifications, let's have the notification area open by default so it's visible. It'll load open instead of collapsed. I remember talking about this with Yaro at one point, but I didn't write it down. We could even add a class on some notifications that will auto hide it after 15 seconds or something like the Gutenburg notice.

File: includes/admin/settings/class-settings-bootstrap.php (hunk 10 above)
Comment (theRealRizeo) [reply includes/admin/settings/class-settings-bootstrap.php]: Spacing reverted in this file

File: includes/admin/class-admin.php (hunk 11 above)
Comment (theRealRizeo) [reply includes/admin/class-admin.php]: Removed extra PHPCS, but diff seems to still be big. Compared with latest version from master and only added tabs on changed files

File: includes/admin/settings/class-settings-bootstrap.php (hunk 10 above)
Comment (stephywells) [reply includes/admin/settings/class-settings-bootstrap.php]: Nevermind! This one is great now.

File: includes/utils.php (hunk 9 above)
Comment (theRealRizeo) [reply includes/utils.php]: This has been fixed :)

File: templates/admin/form-fields.tpl.php (hunk 8 above)
Comment (theRealRizeo) [reply templates/admin/form-fields.tpl.php]: I have fixed the format for this

**Comment by stephywells** [comment]
@semelyak would you mind doing a detailed review of this PR, to make sure that everything looks and works the way you would expect?

@theRealRizeo With a quick glance through, this is looking much better! I've attached a couple screenshots of issues I noticed while I was working on the licensing issue.

In the lite version, this flashes, then disappears:
Screen Shot 2022-01-11 at 3 54 12 PM

Right after the Featured add-on is installed:
Screen Shot

Can you also check saving a license on the Settings -> Licenses page? The positions of messages and buttons move around when a license is saved.

**Comment by semelyak** [comment]
Everything looks good. Just Plans page looks a bit busy. I will modify it

**Comment by garretlaxton** [comment]
I'm seeing some of our pages missing from the WP menu.

Before:
Screenshot 2022-01-13 at 08-41-31 Directory ‹ BD Demo — WordPress

After:
Screenshot 2022-01-13 at 08-41-51 Settings ‹ wordpress — WordPress

Missing Pages:
- Directory Category
- Directory Tags
- Fee Plans
- Form Fields
- Import & Export

**Comment by garretlaxton** [comment]
A few things I noticed so far.

If someone doesn't have the premium module installed, the upgrade notification will show for those users. Can we hide this once any module is activated since they technically aren't on Lite anymore?
Screenshot 2022-01-13 at 11-53-30 Settings ‹ wordpress — WordPress

The missing license notification shows but quickly disappears on the dashboard. Also, should our notification alert the license is missing?
https://www.loom.com/share/0e723a45de834aababa7d54cef81348b

**Comment by stephywells** [comment]
@theRealRizeo That notification in Garret's screenshot needs to show at the top of the page like it does in the current version, rather than in the notifications.

**Comment by theRealRizeo** [comment]
@stephywells got it :) Will make the change on that.  what about the suggestion from Garret for :

> If someone doesn't have the premium module installed, the upgrade notification will show for those users. Can we hide this once any module is activated since they technically aren't on Lite anymore?

**Comment by stephywells** [comment]
> @stephywells got it :) Will make the change on that. what about the suggestion from Garret for :
> 
> > If someone doesn't have the premium module installed, the upgrade notification will show for those users. Can we hide this once any module is activated since they technically aren't on Lite anymore?

@theRealRizeo How about we add this as a separate issue? It's not really related to the changes we are making in this PR.

**Comment by theRealRizeo** [comment]
@garretlaxton This is ready for QA :)

**Comment by theRealRizeo** [comment]
Listing Page

image

Category page

image

Add new category

image


Tags is same as category.

Plans Page

image

Form Fields:

image

Import/Export page

image


Export page :

image


Settings Page

image

Settings Advanced Page

image

Recaptcha

image

Listing Settings

image


Email Settings

image

Payment Settings

image


Appearance Settings

image

Misc Settings

image

Modules:

image

Themes:

image

**Comment by theRealRizeo** [comment]
Notification appearance (red dot pulses) :

image

Pulse notification

https://www.loom.com/share/38cff3eb702948868c4b3cf6734ca455

**Comment by theRealRizeo** [comment]
Fee delete modal. We dont currently assign to the default fee hence the short message

image

**Comment by garretlaxton** [comment]
This looks pretty good so far, just a few minor things.

1. When going to the add-ons page and installing a module, the `Not Installed` status doesn't change to `Active` until the page is refreshed.
Screenshot 2022-01-25 at 08-22-21 Business Directory Modules ‹ wordpress — WordPress

2. If you don't have a zip code DB installed, the notification shows inconsistently.
**Content Page:**
Screenshot 2022-01-25 at 08-26-08 Directory ‹ wordpress — WordPress
**Settings:**
Screenshot 2022-01-25 at 08-39-40 Settings ‹ wordpress — WordPress

Going to keep looking a bit more on this.

**Comment by theRealRizeo** [comment]
Hey @garretlaxton not sure if you are using the latest commit but also try hard refresh the page either using `cmd+f5` mac or `ctrl+f5` windows. This is mainly for the notices

**Comment by theRealRizeo** [comment]
Migrator nav items task https://github.com/Strategy11/business-directory-migrate/issues/4 I think the menu items should be in 1

**Comment by garretlaxton** [comment]
@theRealRizeo Clearing browser cache and viewing in a private window didn't solve the notification issue. I am indeed on the latest commit.

**Comment by theRealRizeo** [comment]
@garretlaxton for Zipcode search, please use the latest master version from the repo. The styling change is mainly for the main plugin, not all addons have been adapted to follow the styling. If there is an issue with an addon it can be ignored, the first step in this is getting the main styling of the plugin working

**Comment by garretlaxton** [comment]
My issue isn't really related to the addons, but accessing them. When I scroll down the page the menu doesn't move. See my video here: https://www.loom.com/share/9798c6fcfd9a4aceb09908f75fed2081

**Comment by theRealRizeo** [comment]
@garretlaxton will check on smaller screen resolutions. Your previous comment mentioned a notification hence my reference to that.

@semelyak I don't think we can keep the nav menu fixed as it will not look good on smaller screens. Will set it to maintain the scroll on the page

**Comment by theRealRizeo** [comment]
@garretlaxton changes mage

**Comment by theRealRizeo** [comment]
Thanks :) @semelyak we need a design to show theme updates available. Currently this is what is shown in 

image

**Comment by theRealRizeo** [comment]
@garretlaxton thanks for testing :) I have pushed some changes:

- Notification of update now shows
- Fixed the delete colour (theme delete text colour) to red
- Fixed the archive icon hover state by replacing it with an admin archive svg icon

**Comment by garretlaxton** [comment]
Looks good, but when I try to update the theme I get this error.
Screenshot 2022-02-08 at 17-58-03 Directory Themes ‹ wordpress — WordPress

**Comment by theRealRizeo** [comment]
Thanks @garretlaxton could be license issue, but its not related to the settings update :)

**Comment by garretlaxton** [comment]
It seems a bit of our Delete/Trash buttons aren't red. I notice the Trash for listings, as well as the Delete button for categories are blue.
Screenshot 2022-02-09 at 02-32-01 Directory ‹ wordpress — WordPress

When creating a category or tag the modal doesn't close after it is created.
Screenshot 2022-02-09 at 02-36-11 Directory Categories ‹ wordpress — WordPress

**Comment by garretlaxton** [comment]
whoops, wrong button... 🥴

File: includes/admin/settings/class-settings-bootstrap.php [hunk 12]
```diff
@@ -914,6 +952,7 @@ private static function settings_appearance() {
                 'name'    => _x( 'Min Image File Size (KB)', 'settings', 'business-directory-plugin' ),
                 'default' => '0',
                 'group'   => 'images/general',
+				'class'   => 'wpbdp-half wpbdp-input-wide',
```
Comment (stephywells) [inline includes/admin/settings/class-settings-bootstrap.php:955]: Can these be changed to wpbdp-col-grid-2? Is wpbdp-input-wide needed?

File: includes/admin/views/modals/field-delete.php [hunk 13]
```diff
@@ -0,0 +1,35 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'You are not allowed to call this page directly.' );
+}
+?>
+<div id="wpbdp-field-delete-modal" class="hidden settings-lite-cta">
```
Comment (stephywells) [inline includes/admin/views/modals/field-delete.php:6]: Can we make a more generic confirmation modal that we can use for plan deleting too? This looks like it could get out of hand to have to add this much new code for every confirmation.

File: includes/class-assets.php [hunk 14]
```diff
@@ -283,13 +290,13 @@ public function enqueue_admin_scripts( $force = false ) {
 
 		wp_enqueue_script( 'wpbdp-frontend-js', WPBDP_ASSETS_URL . 'js/wpbdp.min.js', array( 'jquery' ), WPBDP_VERSION, true );
 
-		wp_enqueue_script( 'wpbdp-admin-js', WPBDP_ASSETS_URL . 'js/admin.min.js', array( 'jquery', 'thickbox', 'jquery-ui-sortable' ), WPBDP_VERSION, true );
+		wp_register_script( 'wpbdp-admin-layout-js', WPBDP_ASSETS_URL . 'js/admin-layout.min.js', array( 'jquery' ), WPBDP_VERSION, true );
```
Comment (stephywells) [inline includes/class-assets.php:293]: Is there a reason to make this a separate file?

**Review by stephywells — COMMENTED** [review]
@theRealRizeo i haven't gotten through everything, but here's a start. Thank you!

File: includes/helpers/class-app.php [hunk 15]
```diff
@@ -317,16 +317,36 @@ public static function show_logo( $size ) {
...
+	/**
+	 * Render a round logo.
+	 *
+	 * @param int $size The logo size.
+	 * @param bool|string $logo_class The optional logo class
+	 *
+	 * @since x.x
+	 */
+	public static function show_round_logo( $size, $logo_class = false ) {
+		echo self::kses( self::svg_logo( $size, array(
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply includes/helpers/class-app.php]: This was changed, the function doesn't look like this anymore :)

File: includes/admin/controllers/class-settings-admin.php [hunk 16]
```diff
@@ -320,17 +390,51 @@ public function setting_multicheck_callback( $setting, $value ) {
...
+		if ( $setting['grid_layout'] ) {
+			$setting['class'] = $setting['class'] . ' wpbdp-grid';
+		}
+
+		echo '<div class="wpdb-select ' . WPBDP_App_Helper::sanitize_html_classes( $setting['class'] ) . '">';
+		if ( $setting['grid_layout'] ) {
+			echo '<div class="' . WPBDP_App_Helper::sanitize_html_classes( $setting['grid_classes']['left'] ) . '">';
+		}
+		echo $this->setting_input_label( $setting, 'div', 'wpbdp-setting-label' );
+		if ( ! empty( $setting['tooltip'] ) ) {
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply includes/admin/controllers/class-settings-admin.php]: Tool tips show instantly on my end when I mouse over. I am using jquery ui. I do not think anything is meant to happen on click. Did you try hard refresh the page?

File: assets/css/less/grids.less [hunk 17]
```diff
@@ -54,6 +58,16 @@
 	grid-column: span 11/span 11;
 }
 
+.wpbdp-grid-mixin-loop (@index) when (@index > 0) {
```
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply assets/css/less/grids.less:61]: This is being used generate grids of different dimensions eg `wpbdp-col-4` `wpbdp-col-8` It reduces the less code needed to generate this

File: includes/admin/settings/class-settings-bootstrap.php (hunk 12 above)
Comment (theRealRizeo) [reply includes/admin/settings/class-settings-bootstrap.php:955]: The `wpbdp-half` aligns the current container into a grid of 50% by 50% . The `wpbdp-col-grid-2` aligns child items in this order. It wont work in this layout

File: includes/admin/controllers/class-settings-admin.php (hunk 16 above)
[the remark this answers is not in the document]
Comment (theRealRizeo) [reply includes/admin/controllers/class-settings-admin.php]: Same code as in formidable to initialize the tooltip is being used

File: includes/admin/views/modals/field-delete.php (hunk 13 above)
Comment (theRealRizeo) [reply includes/admin/views/modals/field-delete.php:6]: Agreed :)

File: includes/class-assets.php (hunk 14 above)
Comment (theRealRizeo) [reply includes/class-assets.php:293]: The main file was getting too big and its a bit all over the place with different duplicate `jQuery(document).ready(function($){` and `(function($) {` called in the file. As a separate file it is easier to find the changes required. But this can be combined into one with a grunt combine config https://www.npmjs.com/package/grunt-combine-files

File: includes/helpers/class-app.php (hunk 15 above)
[the remark this answers is not in the document]
Comment (stephywells) [reply includes/helpers/class-app.php]: Yep. I'm not sure how long that comment was pending. Sorry!

File: assets/css/less/grids.less (hunk 17 above)
[the remark this answers is not in the document]
Comment (stephywells) [reply assets/css/less/grids.less:61]: I see. The Less threw me off.

So .wpbdp8 and .wpbdp-col-8 are duplicate classes then?

[the remark this answers is not in the document]
Comment (theRealRizeo) [reply assets/css/less/grids.less:61]: The `.wpbdp8` is used to set columns with a span of 8/8. For `.wpbdp-col-8` it will only set 1 column. Its sort of set up like bootstrap columns where the max column length is 12 and for positioning the two class combinations for different divs would be `.wpbdp-col-8` and `.wpbdp-col-4`

File: includes/class-assets.php (hunk 14 above)
Comment (theRealRizeo) [reply includes/class-assets.php]: The Grunt build is kind of rigid...maybe later on we can improve this. I have moved the contents to the admin.js file for now

Comment (stephywells) [reply includes/class-assets.php]: Yep. Looks like we need to add a concat option. But for now, I think it's fine to combine them like you did. Thanks!

**Comment by garretlaxton** [comment]
I know you mentioned this isn't really related to the modules, but our region module is the only module where the settings look a little ugly still. Just not sure if something is missing or this needs to be fixed within the module. The new sidebar is also missing from the page. Again, it's only this module.
Screenshot 2022-02-14 at 15-33-59 Directory Regions ‹ wordpress — WordPress

**Comment by theRealRizeo** [comment]
@garretlaxton I will add a separate branch in the regions module to try and make it look better :) Thanks for pointing this out

**Comment by theRealRizeo** [comment]
@garretlaxton please use this branch for regions addon https://github.com/Strategy11/business-directory-regions/tree/interface-update

File: includes/admin/controllers/class-form-fields-admin.php [hunk 18]
```diff
@@ -264,8 +269,11 @@ private function hidden_fields_for_type( $field ) {
 	}
 
 	private function delete_field() {
-		// Check permission.
-		check_admin_referer( 'deletefield' );
+		$nonce = wpbdp_get_var( array( 'param' => 'wpbdp_admin_delete_nonce' ), 'post' );
+		if ( ! $nonce ) {
```
Comment (stephywells) [inline includes/admin/controllers/class-form-fields-admin.php:273]: @theRealRizeo Would you mind moving this nonce check into a different branch so we can get it merged in? Also, I'm not sure why the referer isn't getting checked now if there's no nince.

Comment (stephywells) [reply includes/admin/controllers/class-form-fields-admin.php:273]: The problem is that the nonce can easily be skipped that way. Is there a better way of checking it when we know we need it checked?

Comment (theRealRizeo) [reply includes/admin/controllers/class-form-fields-admin.php:273]: I will check, just thought of an easier way to handle this :)

**Comment by garretlaxton** [comment]
Regions look better. Not a big fan of the orange text in the active tab though, just my opinion!
Screenshot 2022-02-16 at 16-08-49 Directory Regions ‹ wordpress — WordPress

**Comment by theRealRizeo** [comment]
> Regions look better. Not a big fan of the orange text in the active tab though, just my opinion! Screenshot 2022-02-16 at 16-08-49 Directory Regions ‹ wordpress — WordPress

Agreed, would need some design in a later stage for that :)

**Comment by stephywells** [comment]
@theRealRizeo can you please take another look at the tooltips? This is conflicting with the way the tooltips look in Formidable. it's causing js errors:

Uncaught Error: no such method 'show' for tooltip widget instance
    at Function.error (jquery.js?ver=3.6.0:338:9)
    at HTMLInputElement.<anonymous> (core.js?ver=1.13.0:1277:16)
    at Function.each (jquery.js?ver=3.6.0:385:19)
    at jQuery.fn.init.each (jquery.js?ver=3.6.0:207:17)
    at jQuery.fn.init.$.fn.<computed> [as tooltip] (core.js?ver=1.13.0:1260:10)
    at HTMLInputElement.<anonymous> (formidable_admin.js?ver=5.2:553:19)
    at HTMLInputElement.handle (jquery.js?ver=3.6.0:5895:29)
    at HTMLDivElement.dispatch (jquery.js?ver=3.6.0:5430:27)
    at HTMLDivElement.elemData.handle (jquery.js?ver=3.6.0:5234:28)

**Comment by theRealRizeo** [comment]
https://github.com/Strategy11/BusinessDirectoryPlugin/issues/5041 Working on the JS and CSS issue in this branch

File: assets/css/less/grids.less (hunk 17 above)
[the remark this answers is not in the document]
Comment (stephywells) [reply assets/css/less/grids.less]: I just tested this, and they are the same thing. I went ahead and removed the duplicates.

File: includes/admin/controllers/class-admin-fees.php [hunk 19]
```diff
@@ -88,16 +119,18 @@ private function insert_or_update_fee( $mode ) {
...
 
-		$images_changed = false;
 		if ( 'insert' === $mode ) {
             $fee = new WPBDP__Fee_Plan( $posted_values );
+			$images_changed = false;
         } else {
-			$fee            = $this->get_or_die();
-			$images_changed = (int) $fee->images !== (int) $posted_values['images'];
+			$fee = $this->get_or_die();
+			$images_changed = $images !== false && (int) $fee->images !== $images;
```
Comment (stephywells) [inline includes/admin/controllers/class-admin-fees.php:133]: @theRealRizeo The changes lines from 122-133 don't look related to this PR. Can you please move this out to a new branch? I'm not sure what this is for.

Comment (theRealRizeo) [reply includes/admin/controllers/class-admin-fees.php:138]: Hey @stephywells It' a change you committed https://github.com/Strategy11/business-directory-plugin/commit/58a9e591102cfef35c23160f554e7495c7d2fdf3 I will revert it

Comment (theRealRizeo) [reply includes/admin/controllers/class-admin-fees.php:138]: Last commit reverted to fix for undefined variable
